src/olympia/reviewers/views.py

```diff
@@ -1374,7 +1374,7 @@ def retrieve(self, request, *args, **kwargs):
                     'request': self.request
                 }
             )
-            return Response({'file': serializer.data})
+            return Response({'id': version.id, 'file': serializer.data})
```

Comment:
To use DRF patterns shouldn't be building up responses like this - this is what a Serializer is for.  `AddonBrowseVersionSerializer` should be subclassed (or a parent class introduced); `get_serializer()` overridden on `ReviewAddonVersionViewSet`, and `get_serializer_context` overridden on `ReviewAddonVersionViewSet` to provide the context.  - Then we shouldn't need to redefine `retrieve` because that's what the `RetrieveModelMixin` is supposed to be doing.

Reply:
Thanks @eviljeff, that makes sense. I'll have to take a closer look at the `retrieve` thing, though, because this view already redefined `retrieve` so there must have been a need for that.

Reply:
The `RetrieveModelMixin` literally just defines the `retrieve` method [1] so I'm not sure why it was used if was also defined tbh.

[1] https://github.com/encode/django-rest-framework/blob/3.11.0/rest_framework/mixins.py#L49